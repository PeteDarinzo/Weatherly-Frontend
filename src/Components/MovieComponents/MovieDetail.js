import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useParams, useHistory } from "react-router-dom";
import { fetchMovieFromAPI } from "../../Actions/actions";
import { deleteFromWatchList } from "../../Actions/actions";
import Container from '@mui/material/Container';
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid';
import Stack from '@mui/material/Stack';
import Button from '@mui/material/Button';
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles({
  container: {
    padding: 40
  }
});


const MovieDetail = () => {

  const classes = useStyles();

  const { movieId } = useParams();
  let movie = useSelector(store => store.movies[movieId]);

  const dispatch = useDispatch();
  const history = useHistory();

  useEffect(() => {

    async function getMovie() {
      await dispatch(fetchMovieFromAPI(movieId));
    }

    if (!movie) {
      getMovie();
    }

  }, [dispatch, movieId, movie]);

  function handleClick() {
    dispatch(deleteFromWatchList("peter", movieId));
    history.push("/");
  }


  if (!movie) return (<b>Loading...</b>)

  return (
    <Container className={classes.container}>
      <Grid container>
        <Grid item md={4}>
          <img width="250px" src={movie.Poster} />
        </Grid>
        <Grid item md={4}>
          <Stack spacing={2}>
            <Typography variant="h3">{movie.Title}</Typography>
            <Typography><b>Year:</b> {movie.Year}</Typography>
            <Typography><b>Genre:</b> {movie.Genre}</Typography>
            <Typography><b>IMDB Rating:</b> {movie.imdbRating}</Typography>
            <Typography><b>Length:</b> {movie.Runtime}</Typography>
            <Typography><b>Plot:</b> {movie.Plot}</Typography>
            <Button
              onClick={handleClick}
              variant="contained"
              color="error"
            >
              Remove
            </Button>

          </Stack>
        </Grid>
      </Grid>
    </Container>
  );
}

export default MovieDetail;