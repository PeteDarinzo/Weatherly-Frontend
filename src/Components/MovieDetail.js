import React from "react";
import Button from '@mui/material/Button';


const MovieDetail = ({ title, year, rating, runtime, genre, plot, poster, }) => {

  return (
    <div>
      <img src={poster} />
      <h1>{title}</h1>
      <p>{year}</p>
      <p>{genre}</p>
      <p>{rating}</p>
      <p>{runtime}</p>
      <p>{plot}</p>
      <Button variant="contained" color="error">
        Remove
      </Button>
    </div>
  );
}

export default MovieDetail;