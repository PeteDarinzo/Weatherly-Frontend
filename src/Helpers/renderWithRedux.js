import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}