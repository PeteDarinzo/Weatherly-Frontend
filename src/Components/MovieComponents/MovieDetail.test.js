import MovieDetail from "./MovieDetail";
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import renderWithRedux from "../../Helpers/renderWithRedux";
import { rest } from "msw";
import { setupServer } from "msw/node";

const server = setupServer(
  rest.get("http://localhost:3001/movies/id", (req, res, ctx) => {
    return res(
      ctx.json({
        Title: "test",
        Year: "1999",
        Genre: "action",
        imdbRating: "10",
        Runtime: "60",
        Plot: "test plot"
      })
    )
  }
  ))


it("renders movie detail without crashing", () => {
  renderWithRedux(
    <MemoryRouter initialEntries={["/movies/id/1"]}>
      <MovieDetail removeMovie={() => { }} />
    </MemoryRouter>
  );
});


it("matches the movie detail snapshot", () => {
  const { asFragment } = renderWithRedux(
    <MemoryRouter>
      <MovieDetail removeMovie={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
