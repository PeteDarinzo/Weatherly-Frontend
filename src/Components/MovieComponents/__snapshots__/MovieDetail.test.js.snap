// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the movie detail snapshot 1`] = `
<DocumentFragment>
  <b>
    Loading...
  </b>
</DocumentFragment>
`;
